src/modelfree/configs/multi/train.py

```diff
@@ -95,42 +274,213 @@ def hyper(train):
+        best policy from hyper_finetune_defense"""
+        train = dict(train)
+        spec = _ysnp_hyper_against_finetune_defense(train)
+        train['victim_path'] = osp.join(
+                                MULTI_TRAIN_LOCATION,
+                                YSNP_DEFENSE_POLICIES_LOOKUP['adversary_train']['best_hyper_3e6'])
+
+        train['load_policy'] = {
+            'path': None,
+            'type': "ppo2"
+        }
+        exp_name = 'hyper_against_adv_finetuned_from_scratch'
+        _ = locals()  # quieten flake8 unused variable warning
+        del _
+
+    @multi_train_ex.named_config
+    def hyper_against_dual_finetuned_from_scratch(train):
+        """HP search for training adversary from scratch against best policy from
+        hyper_finetune_dual_defense """
+        train = dict(train)
+        spec = _ysnp_hyper_against_finetune_defense(train)
+        train['victim_path'] = osp.join(
+                                MULTI_TRAIN_LOCATION,
+                                YSNP_DEFENSE_POLICIES_LOOKUP['dual_train']['best_hyper_3e6'])
+        train['load_policy'] = {
+            'path': None,
+            'type': "ppo2"
+        }
+        exp_name = 'hyper_against_dual_finetuned_from_scratch'
+        _ = locals()  # quieten flake8 unused variable warning
+        del _
+
+    @multi_train_ex.named_config
+    def hyper_against_adv_finetuned_from_existing(train):
+        """ HP search for finetuning adversary from existing adversary against
+        best policy from hyper_finetune_defense"""
+        train = dict(train)
+        spec = _ysnp_hyper_against_finetune_defense(train)
+        train['victim_path'] = osp.join(
+            MULTI_TRAIN_LOCATION,
+            YSNP_DEFENSE_POLICIES_LOOKUP['adversary_train']['best_hyper_3e6'])
+
+        train['load_policy'] = {
+            'path': osp.join(MULTI_TRAIN_LOCATION, YSNP_ADVERSARY),
+            'type': "ppo2"
+        }
+        exp_name = 'hyper_against_adv_finetuned_from_existing'
+        _ = locals()  # quieten flake8 unused variable warning
+        del _
+
+    @multi_train_ex.named_config
+    def hyper_against_dual_finetuned_from_existing(train):
+        """ HP search for finetuning adversary from existing adversary against
+                best policy from hyper_finetune_dual_defense"""
+        train = dict(train)
+        spec = _ysnp_hyper_against_finetune_defense(train)
+        train['victim_path'] = osp.join(
+            MULTI_TRAIN_LOCATION,
+            YSNP_DEFENSE_POLICIES_LOOKUP['dual_train']['best_hyper_3e6'])
+
+        train['load_policy'] = {
+            'path': osp.join(MULTI_TRAIN_LOCATION, YSNP_ADVERSARY),
+            'type': "ppo2"
+        }
+        exp_name = 'hyper_against_dual_finetuned_from_existing'
+        _ = locals()  # quieten flake8 unused variable warning
+        del _
+
+    # BEST-HYPERPARAM LONG RUN: RETRAIN ADVERSARY
+
+    @multi_train_ex.named_config
+    def ysnp_adversary_retraining_adv_v0(train):
```

Comment:
Which ones do you want to remove? It's not actually clear to me which are the least essential, since I think we'd need to rerun basically all of them to validate our earlier results. I guess we could remove the long-run configs, since those e.g. learning rates might change, but we'll just end up having to add them back in with different learning rates once we finish the rounds of hyperparameter tuning

Reply:
So, do you want to get rid of all of the long-run experiments, then, because the hyperparameter values for those might change?